paths_cli/plugin_management.py

```diff
@@ -3,62 +3,149 @@
 import importlib
 import os
 
-OPSPlugin = collections.namedtuple(
-    "OPSPlugin", ['name', 'location', 'func', 'section', 'plugin_type']
-)
+class PluginRegistrationError(RuntimeError):
+    pass
+
+# TODO: make more generic than OPS (requires_ops => requires_lib)
+class Plugin(object):
+    """Generic plugin object
+
+    parameters
+    ----------
+    requires_lib: tuple
+        tuple representing the minimum allowed version of the underlying
+        library
+    requires_cli: tuple
+        tuple representing hte minimum allowed version of the command line
+        interface application
+    """
+    def __init__(self, requires_lib, requires_cli):
+        self.requires_lib = requires_lib
+        self.requires_cli = requires_cli
+        self.location = None
+        self.plugin_type = None
+
+    def attach_metadata(self, location, plugin_type):
+        # error is already registered and data doesn't match
+        error_condition = (
+            (self.location is not None or self.plugin_type is not None)
+            and (self.location != location
+                 or self.plugin_type != plugin_type)
+        )
+        if error_condition:  # -no-cov-
+            raise PluginRegistrationError(
+                "The plugin " + repr(self) + "has been previously "
+                "registered with different metadata."
```

Comment:
f string possible (also added a whitespace after the `repr`)
```suggestion
                f"The plugin {repr(self)} has been previously "
                "registered with different metadata."
```